codice/ddf
# DDF-04481 Removed Realm support from Web Context Policy

#### What does this PR do?
- Removed the realm attribute from the web context policy manager, authentication tokens, and token validators.
- Deleted the UPBSTValidator that relies heavily on JAAS realms.
- Combined the KarafLogoutAction and LdapLogoutAction into the DefaultLogoutAction and changed the logic of the LogoutService.
- Combined the StsRealm and the WssStsRealm
- Combined the BasicAuthenticationHandler and the WssBasicAuthenticationHandler
- Combined the PKIHandler and the WssPKIHandler.
- Created the BaseAuthenticationTokenFactory to create a token from a username and password and from certificates
- Removed the UPAuthenticationToken and the PKIAuthenticationToken which are no longer needed
- Removed the PKITokenValidator - moving forward tokens with a certificate on them will be validated using the X509PathTokenValidator.
- Removed the realm-specific logout page
- Removed the login sidebar in Intrigue which relied on realms
- Made sure BASIC is a terminating authentication type (if the handler is reached and credentials are not found, it will prompt the user)

#### Who is reviewing it? 
@bakejeyner 

#### Select relevant component teams: 
@codice/security 

#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler
@brjeter
@garrettfreibott
@tbatie 
@ricklarsen - Documentation
@rzwiefel
@stustison

#### How should this be tested?
- **Requires the atlas PR: https://github.com/connexta/atlas/pull/6**
- The default configuration of the Web Context Policy Manager (WCPM) is `/=IDP|GUEST` without changing it test on both the admin console and Intrigue
    - Login and log out with a username and password
    - Login and log out with a certificate
    - Login and log out as guest
    - NOTE: Logout will display a message to close the browser but it shouldn’t be required
- Change the WCPM auth type config to `/=BASIC|GUEST` and test the following on both the admin console and Intrigue
    - You should be prompted for credentials immediately 
    - Login and log out 
        - Logout should require you to close the browser
    - Make sure you are not able to log in as a guest
- Connect to LDAP and verify login in and logging out
- Note: the `SAML|GUEST` configuration will be “broken” when accessing intrigue since SAML is not a terminating authentication type
    - It should display Intrigue for the guest user but will have no way to log in just a Log Out button
- Add the admin wizards and test that they work as expected (with this admin console PR https://github.com/connexta/admin-console/pull/234)

#### What are the relevant tickets?
For GH Issues:
Fixes: https://github.com/codice/ddf/issues/4481

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
